### PR TITLE
Fix ecmwf

### PR DESCRIPTION
**Pull request by bbuzz31**
ERA5 failed during cropping of the weather model to the user specified AOI due to changes in how the coordinate arrays are set up (#538).

I've corrected the shape broadcasting for ERA5 to pass, and verified that delays produced on this PR for ERA5, HRES, GMAO and HRRR are equivalent to delays produced on dev prior to #538  (#525).

Tests pass locally.

### CODE REVIEW

File: tools/RAiDER/models/ecmwf.py [hunk 1]
```diff
@@ -101,11 +101,10 @@ def _load_model_level(self, fname):
         self._q = q
         geo_hgt, pres, hgt = self._calculategeoh(z, lnsp)
 
-        # re-assign lons, lats to match heights
-        _lons = np.broadcast_to(lons[np.newaxis, np.newaxis, :], hgt.shape)
-        _lats = np.broadcast_to(lats[np.newaxis, :, np.newaxis], hgt.shape)
+        self._lons, self._lats = np.meshgrid(lons, lats)
```
Comment (bbuzz31) [inline tools/RAiDER/models/ecmwf.py:104]: this is needed for consistency with HRRR; a 2D array should be used

File: tools/RAiDER/models/ecmwf.py [hunk 2]
```diff
@@ -101,11 +101,10 @@ def _load_model_level(self, fname):
...
+
         # ys is latitude
-        self._get_heights(_lats, hgt)
+        self._get_heights(self._lats, hgt.transpose(1, 2, 0))
```
Comment (bbuzz31) [inline tools/RAiDER/models/ecmwf.py:107]: this, I thought, should make the function work as internally in _get_heights we add a dimension to the last axis and then stack the lat/lons in height

File: tools/RAiDER/models/weatherModel.py [hunk 3]
```diff
@@ -324,7 +324,10 @@ def _get_heights(self, lats, geo_hgt, geo_ht_fill=np.nan):
         Transform geo heights to WGS84 ellipsoidal heights
         '''
         geo_ht_fix = np.where(geo_hgt != geo_ht_fill, geo_hgt, np.nan)
-        lats_full = np.broadcast_to(lats[...,np.newaxis], geo_ht_fix.shape)
+        print ('LATS SHAPE', lats.shape)
```
Comment (bbuzz31) [inline tools/RAiDER/models/weatherModel.py:327]: some diagnostics to show the shapes; HRRR conus (which works) and ERA5 are aligned dimension wise but the latter doesn't work

File: tools/RAiDER/models/ecmwf.py (hunk 1 above)
Comment (piyushrpt) [reply tools/RAiDER/models/ecmwf.py:104]: You don't necessarily have to use 2D. If memory is limited or you want to keep footprint down, use numpy stride tricks:

https://towardsdatascience.com/advanced-numpy-master-stride-tricks-with-25-illustrated-exercises-923a9393ab20

File: tools/RAiDER/models/weatherModel.py [hunk 4]
```diff
@@ -324,7 +324,10 @@ def _get_heights(self, lats, geo_hgt, geo_ht_fill=np.nan):
...
+        print (geo_ht_fix.shape)
+        lats_full  = np.broadcast_to(lats[...,np.newaxis], geo_ht_fix.shape)
```
Comment (bbuzz31) [inline tools/RAiDER/models/weatherModel.py:329]: this line is needed for HRRR

**Review by jlmaurer — APPROVED** [review]
LGTM as long as relevant tests pass

**Comment by jlmaurer** [comment]
@bbuzz31 do we have any tests that test with ECMWF models currently?

**Comment by bbuzz31** [comment]
@jlmaurer some of the tests have a variable (WM) that can be changed to ERA-5. Sometimes ERA-5 takes many hours to download.

In #543 I'm reusing downloaded ERA5 models for `test_intersect.py` so that is a partial test.

**Comment by jlmaurer** [comment]
Ok great. Do those tests pass? The problem here is one of ordering.

**Comment by bbuzz31** [comment]
Yup they pass

**Pull request merged**